test/e2e/tests/test_monitoring_schedule.py

```diff
@@ -0,0 +1,219 @@
+        MonitoringScheduleName=monitoring_schedule_name
+    )["MonitoringScheduleStatus"]
+
+
+def get_monitoring_schedule_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    assert "monitoringScheduleStatus" in resource["status"]
+    return resource["status"]["monitoringScheduleStatus"]
+
+
+def wait_sagemaker_monitoring_schedule_status(
+    sagemaker_client,
+    monitoring_schedule_name,
+    expected_status: str,
+    wait_periods: int = 6,
+    period_length: int = 30,
+):
+    return wait_for_status(
+        expected_status,
+        wait_periods,
+        period_length,
+        get_monitoring_schedule_sagemaker_status,
+        sagemaker_client,
+        monitoring_schedule_name,
+    )
+
+
+def wait_resource_monitoring_schedule_status(
+    reference: k8s.CustomResourceReference,
+    expected_status: str,
+    wait_periods: int = 6,
+    period_length: int = 30,
+):
+    return wait_for_status(
+        expected_status,
+        wait_periods,
+        period_length,
+        get_monitoring_schedule_resource_status,
+        reference,
+    )
+
+
+@service_marker
+@pytest.mark.canary
+class TestMonitoringSchedule:
+    status_pending: str = "Pending"
+    status_scheduled: str = "Scheduled"
+
+    def _assert_monitoring_schedule_status_in_sync(
+        self,
+        sagemaker_client,
+        schedule_name,
+        reference,
+        expected_status,
+        wait_periods: int = 6,
+        period_length: int = 30,
+    ):
+        assert (
+            wait_sagemaker_monitoring_schedule_status(
+                sagemaker_client, schedule_name, expected_status
+            )
+            == wait_resource_monitoring_schedule_status(reference, expected_status, 2)
+            == expected_status
+        )
+
+    def test_smoke(self, sagemaker_client, xgboost_churn_data_quality_monitoring_schedule):
+        (reference, resource, spec) = xgboost_churn_data_quality_monitoring_schedule
+        assert k8s.get_resource_exists(reference)
+
+        monitoring_schedule_name = resource["spec"].get("monitoringScheduleName")
+
+        assert (
+            k8s.get_resource_arn(resource)
+            == describe_sagemaker_monitoring_schedule(
+                sagemaker_client, monitoring_schedule_name
+            )["MonitoringScheduleArn"]
+        )
+
+        # scheule transitions Pending -> Scheduled state
+        # Pending status is shortlived only for 30 seconds because baselining job has already been run
+        # remove the checks for Pending status if the test is flaky because of this
+        # as the main objective is to test for Scheduled status
+        # OR
+        # create the schedule with a on-going baseline job where it waits for the baselining job to complete
+        assert (
+            wait_resource_monitoring_schedule_status(
+                reference, self.status_pending, 5, 2
+            )
+            == self.status_pending
```

Comment:
Are we certain it will be in pending? It could skip over pending if we don't poll it immediately?

Reply:
yes and these checks need to happen in 30 seconds
will see and iterate on this